server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1434,6 +1434,14 @@ public final boolean shouldProcessCheckpoint(ReplicationCheckpoint requestCheckp
             logger.trace(() -> new ParameterizedMessage("Ignoring new replication checkpoint - shard is not started {}", state()));
             return false;
         }
+        if (getReplicationTracker().isPrimaryMode()) {
+            logger.trace(
+                () -> new ParameterizedMessage(
+                    "Ignoring new replication checkpoint - shard is in primaryMode and cannot receive any checkpoints."
+                )
+            );
```

Comment:
I am concerned about the silent logging, ideally this should be rare, if we are seeing this quite often, it might reflect a bug. Let log with a WARN level

Reply:
Sure, I will change this to warn level.